game/kernel/jak1/kscheme.cpp

```diff
@@ -470,6 +479,9 @@ Ptr<Function> make_function_from_c(void* func, bool arg3_is_pp = false) {
   return make_function_from_c_linux(func, arg3_is_pp);
 #elif _WIN32
   return make_function_from_c_win32(func, arg3_is_pp);
+#elif __APPLE__
+  // FIXME - reusing linux for now
+  return make_function_from_c_linux(func, arg3_is_pp);
```

Comment:
DRY, but also, linux and macOS are different beasts. Is there any reason to believe this two versions should always be the same (for non-arm at least)? Otherwise perhaps it's better to copy-paste the implementation but call it `make_function_from_c_macos`.

Reply:
@Ziemas sorry for dragging you back, but I thought maybe you know here as well: is the calling convention the reason linux and win32 have different implementations of `make_function_from_c_XXX` and `make_stack_arg_function_from_c_XXX` as well, you think?

If so the linux ones could be renamed to `XXX_from_c_systemv`.

Reply:
Yes, I think that's right. Those can be shared as well.